config/src/main/java/com/typesafe/config/ConfigFactory.java

```diff
@@ -450,7 +450,7 @@ public static Config defaultOverrides() {
      * @return the default override configuration
      */
     public static Config defaultOverrides(ClassLoader loader) {
-        return systemProperties();
+        return systemProperties().withFallback(systemEnvironment());
```

Comment:
It turned out that system properties were used as a default fallback, but the system environment was only used for resolving references. This change now also includes them in the configuration